### DISCUSSION TITLE
KeyError: "Invalid keys:

### DISCUSSION THREAD

**Discussion by opalmeira** [fabiocaccamo/python-benedict#121] [Polls]
Hello,
First I would like to thank you for sharing your program to handle dictionaries.
I installed python-benedict as suggested and am trying to use to manage my complex dictionary which has dictionaries of dictionaries which in turn has lists of dictionaries.
In order to test the program I created a dummy dictionary that is sort of a very simplified version of the target one:
`d = {'a':{'b': {'c':[{'z':['e','f']},2,3], 'd':[5,6,7]}}}`
d = benedict(df)
And I want to retrieve the value like this:
`d["a.b.c[0].z[0]"]`
output: 'e'
This works perfectly, but when I tried on my target dictionary I get the error.
My dictionary is loaded like this:
`my_dict = pd.read_json("path/to/dictionary.json")`
It is actually a json file, so I also tried:
`benedict.from_json`
But it does not work either
If anyone can help me, I would appreciate it.
Thank you!

**Comment by opalmeira** [comment]
Hello, 
I'm back. So I managed to solve the problem. 
I think the issue was because I loaded the json file with pandas. Somehow it does not work if we try to benedict the dictionary. So I reloaded the json file with `benedict.from_json`  using `keypath_separator=None,` then `standardize` the dictionary and last I set up the key separator: `my_dict = benedict(my_dict,keypath_separator = ".")`.
Now it works fine.
If you guys have any comments, please share.
Thank you!

**Comment by fabiocaccamo** [comment]
Hello @opalmeira, 
yes, the reason is that you initialized `benedict` with a `panda`'s object which is not a `dict` but a `dataframe`.

If you want to do it, you should convert the `dataframe` object to `dict` before passing it to `benedict`:
```python
my_df = pd.read_json("path/to/dictionary.json")
my_dict = my_df.to_dict()
my_bd = benedict(my_dict)
```

or you could directly create the `benedict` instance from the .json file:
```python
my_bd = benedict.from_json("path/to/dictionary.json")
```
or simply:
```python
my_bd = benedict("path/to/dictionary.json")
```
